cpp/arcticdb/codec/python_bindings.cpp

```diff
@@ -0,0 +1,226 @@
+            for (std::size_t j = 0; j < dim; ++j, ++s) {
+                v *= *s;
+            }
+            item_count += v;
+        }
+        util::check_arg(item_count == std::size_t(d_info_.size),
+                        "number of elements e={} and sum of shapes s={} do not match",
+                        item_count, d_info_.size);
+
+        auto data_bytes = d_info_.size * d_info_.itemsize;
+        auto shape_bytes = s_info_.size * s_info_.itemsize;
+
+        if (td.dimension() == Dimension::Dim0) {
+            data_.ensure_bytes(data_bytes);
+            memcpy(data_.cursor(), d_info_.ptr, data_bytes);
+            field_ = std::make_shared<ColumnData>(&data_.buffer(), &shapes_.buffer(), td_, nullptr);
+        } else {
+            data_.ensure_bytes(data_bytes);
+            shapes_.ensure_bytes(shape_bytes);
+            memcpy(shapes_.cursor(), s_info_.ptr, shape_bytes);
+            memcpy(data_.cursor(), d_info_.ptr, data_bytes);
+            data_.commit();
+            shapes_.commit();
+            field_ = std::make_shared<ColumnData>(&data_.buffer(), &shapes_.buffer(), td_, nullptr);
+        }
+    }
+
+    std::shared_ptr<ColumnData> as_field() {
+        return field_;
+    }
+
+  private:
+    CursoredBuffer<Buffer> shapes_;
+    CursoredBuffer<ChunkedBuffer> data_;
+    TypeDescriptor td_;
+    std::shared_ptr<ColumnData> field_;
+};
+
+struct FieldEncodingResult {
+    FieldEncodingResult() = default;
+    FieldEncodingResult(std::shared_ptr<Buffer> buffer, proto::encoding::EncodedField encoded_field):
+    buffer_(std::move(buffer)),
+    encoded_field_(encoded_field) {}
+    std::shared_ptr<Buffer> buffer_;
+    proto::encoding::EncodedField encoded_field_;
+};
+
+FieldEncodingResult py_encode_field(py::object &opts, DynamicFieldBuffer &field_buffer) {
+    proto::encoding::VariantCodec opts_cpp;
+    python_util::pb_from_python(opts, opts_cpp);
+
+    std::ptrdiff_t pos = 0;
+    proto::encoding::EncodedField encoded_field;
+    auto field = field_buffer.as_field();
+    const auto [uncompressed, required] = ColumnEncoder::max_compressed_size(opts_cpp, *field);
+    auto buffer = std::make_shared<Buffer>(required);
+    field->reset();
+    ColumnEncoder::encode(opts_cpp, *field, encoded_field, *buffer, pos);
+    return FieldEncodingResult{buffer, encoded_field};
+}
+
+Segment encode_segment(SegmentInMemory segment_in_memory, const py::object &opts) {
+    proto::encoding::VariantCodec opts_cpp;
+    python_util::pb_from_python(opts, opts_cpp);
+    return encode(std::move(segment_in_memory), opts_cpp);
+}
+
+SegmentInMemory decode_segment(Segment segment) {
+    return decode(std::move(segment));
+}
+
+class BufferPairDataSink {
+  public:
+    BufferPairDataSink() :
+        values_(std::make_shared<Buffer>()),
+        shapes_(std::make_shared<Buffer>()) {
+    }
+
+    /**
+     * @param size
+     * @return a pointer to the first byte of a contiguous memory chunk of size bytes
+     */
```

Comment:
Wrong docstring style. Either remove or change from javadoc style

Reply:
This seems to be Doxygen-compatible and works well in most IDEs.

Reply:
Removed